general/g.mapsets/main.c

```diff
@@ -130,7 +163,27 @@ int main(int argc, char *argv[])
         }
     }
 
-    fs = G_option_to_separator(opt.fs);
+    if (strcmp(opt.format->answer, "json") == 0)
+        format = JSON;
+    else
+        format = PLAIN;
+    if (format == JSON) {
+        fatal_error_option_value_excludes_option(
+            opt.format, opt.fsep, _("Separator is part of the format"));
+    }
+
+    /* the field separator */
+    if (opt.fsep->answer) {
+        fsep = G_option_to_separator(opt.fsep);
+    }
+    else {
+        /* A different separator is needed to for each format and output. */
+        if (format == PLAIN) {
+            fsep = G_store("|");
```

Comment:
You are changing the default separator for plain, there should be space to avoid any changes.

Reply:
I set it to `sep = G_store("space");` let me know if that isn't what you meant.